### ISSUE TITLE
tool installation interface bugs and improvements

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#4076]
xref: https://github.com/galaxyproject/galaxy/pull/3626

~~1. (bug) clicking on `Repository Queue (0)` emit JS error `Unexpected token u in JSON at position 0`~~

1. (ux) when searching for a repository I get a list of results such as
![screenshot 2017-04-24 14 22 37](https://cloud.githubusercontent.com/assets/1814954/25352301/9a68ab32-28f9-11e7-8c24-5d7caf518f17.png)
but when I navigate results with arrows (as suggested) the cursor is not highlighted and the name of the repository is replaced by the ID which makes the search experience really confusing. 
![screenshot 2017-04-24 14 22 44](https://cloud.githubusercontent.com/assets/1814954/25352427/fe0913c0-28f9-11e7-9150-a9b0637c9c4f.png)

**Comment by mvdbeek** [comment]
Also the default version of a repository is not always the latest, i.e samtools calmd on the main TS is first picked up with changeset 0, while changeset 1 is newer.

**Comment by martenson** [comment]
Adding to what Marius said the list is not ordered ![screenshot 2017-08-11 15 04 51](https://user-images.githubusercontent.com/1814954/29228126-9c259284-7ea6-11e7-8330-f55752f12540.png)

And when selecting non-default revision an empty page is generated on select breaking the whole interface. JS console has:

```
Uncaught TypeError: Cannot read property 'tool_shed' of undefined
    at N.d.initialize (repository-view.js:22)
    at N.d.reDraw (repository-view.js:208)
    at HTMLSelectElement.<anonymous> (repository-view.js:65)
    at HTMLSelectElement.dispatch (jquery.js:5226)
    at HTMLSelectElement.m.handle (jquery.js:4878)
```

**Comment by martenson** [comment]
@davebx Could you please find time to check on the reported bugs above?

**Comment by davebx** [comment]
@martenson yes, this is now my priority

**Comment by mvdbeek** [comment]
The revision being unordered should be fixed by https://github.com/galaxyproject/galaxy/pull/4944,
but adding to that I also noticed that the "Create new" button doesn't do anything now. Also the layout is a bit off:
![screen shot 2017-11-06 at 17 15 32](https://user-images.githubusercontent.com/6804901/32451043-2e2b5390-c316-11e7-9cdf-4cb143281bad.png)

**Comment by mvdbeek** [comment]
Some more things:
- When you navigate to a repository through the categories, then search for another repository and select the other repository you are presented with the first (wrong) repository to install, and only a page refresh will restore this. 

- Repository queue updates are not rendered (they only appear when you navigate to a new page ...)

**Comment by martenson** [comment]
'Install all tools missing for a workflow'-button would be very nice.

**Comment by martenson** [comment]
17.09: the 'remove from queue' button does not work, and there is no 'install all in the queue' button.

**Comment by martenson** [comment]
@davebx Do you have any WIP on these? I feel a bit bad that `enable_beta_ts_api_install` is true by default since we have this list of bugs.

**Comment by davebx** [comment]
#6544 should fix the empty queue issue.

I'm working on the create new/select existing switcher, it looks like I never ported that over from the `browse_toolsheds.mako` into the client/scripts/mvc version.

**Comment by davebx** [comment]
The tool panel switcher should be fixed as of 99b2e28a9e38434755641a904688d9d79738f061 in the same pull request.

**Comment by martenson** [comment]
work continues here: https://github.com/galaxyproject/galaxy/projects/13
